src/run/emulator.rs

```diff
@@ -1654,6 +1697,88 @@ impl Machine {
                             Err(ExecutionError::new("invalid operand to EcPairing instruction", &self.state, None))
                         }
 
+                    }
+                    Opcode::AVMOpcode(AVMOpcode::NewBuffer) => {
+                        let _size = self.stack.pop_uint(&self.state)?;
+                        self.stack.push(Value::new_buffer(vec![0; 256]));
+                        self.incr_pc();
+                        Ok(true)
+                    }
+                    Opcode::AVMOpcode(AVMOpcode::GetBuffer8) => {
+                        let buf = self.stack.pop_buffer(&self.state)?;
+                        let offset = self.stack.pop_usize(&self.state)?;
+                        self.stack.push_usize(buf.read_byte(offset).into());
+                        self.incr_pc();
+                        Ok(true)
+                    }
+                    Opcode::AVMOpcode(AVMOpcode::GetBuffer64) => {
+                        let buf = self.stack.pop_buffer(&self.state)?;
+                        let offset = self.stack.pop_usize(&self.state)?;
+                        if offset + 7 < offset {
+                            return Err(ExecutionError::new("buffer overflow", &self.state, Some(Value::Int(Uint256::from_usize(offset)))))
+                        }
+                        let mut res = [0u8; 8];
+                        for i in 0..8 {
+                            res[i] = buf.read_byte(offset+i);
+                        }
+                        self.stack.push_uint(Uint256::from_bytes(&res));
+                        self.incr_pc();
+                        Ok(true)
+                    }
+                    Opcode::AVMOpcode(AVMOpcode::GetBuffer256) => {
+                        let buf = self.stack.pop_buffer(&self.state)?;
+                        let offset = self.stack.pop_usize(&self.state)?;
+                        if offset + 31 < offset {
+                            return Err(ExecutionError::new("buffer overflow", &self.state, Some(Value::Int(Uint256::from_usize(offset)))))
+                        }
+                        let mut res = [0u8; 32];
+                        for i in 0..32 {
+                            res[i] = buf.read_byte(offset+i);
+                        }
+                        self.stack.push_uint(Uint256::from_bytes(&res));
+                        self.incr_pc();
+                        Ok(true)
+                    }
+                    Opcode::AVMOpcode(AVMOpcode::SetBuffer8) => {
+                        let buf = self.stack.pop_buffer(&self.state)?;
```

Comment:
Suggest having the order of args on the stack be (top to bottom): offset, val, buf.   Offset is must likely to be constant, so put it on top--and that's consistent with the GetBuffer instructions.  Then put val next to that it's efficient to do a series of SetBuffers in a row by pushing the buffer, then repeating the sequence push val, push offset.